### ISSUE TITLE
Can't change req.headers

### ISSUE THREAD

**Issue by maxpain** [dimdenGD/ultimate-express#66]
Hello. I'm migrating from legacy `express` to `ultimate-express` and this code doesn't work in `ultimate-express`:

```typescript
import type { Request, NextFunction, Response } from 'ultimate-express'

export default async function hasuraHeadersMiddleware(
	request: Request,
	response: Response,
	next: NextFunction,
): Promise<void> {
	request.headers.cookie = 'test'
	console.log('cookie header:', request.headers.cookie) // cookie header: undefined

	next()
}
```

The `cookie` header in `request` isn't changing for some reason.
This code works in `express` by the way.

**Comment by nigrosimone** [comment]
In `ultimate-express`  the `request.headers` is a getter and the headers are cached for maximum performance, see https://github.com/dimdenGD/ultimate-express/blob/main/src/request.js#L348

Why you need to change the request header ?

The workaround is change headers into `request.#cachedHeaders` but `cachedHeaders` is marked as private field

**Comment by maxpain** [comment]
In my case, I have a middleware for the auth webhook for the Hasura GraphQL engine.
It gets headers from a JSON body and overwrites `req.headers`, so `cookie-parser` could get `req.headers.cookie`.

```typescript
import type { Request, NextFunction, Response } from 'ultimate-express'

export default async function hasuraHeadersMiddleware(
	request: Request,
	response: Response,
	next: NextFunction,
): Promise<void> {
	const { headers } = request.body as { headers: Record<string, string> }

	if (request.path === '/hasura_auth' && headers) {
		for (const key in headers) {
			if (Object.hasOwn(headers, key)) {
				request.headers[key.toLowerCase()] = headers[key]!
			}
		}
	}

	next()
}
```

```typescript
app.use(express.json())
app.use(hasuraHeadersMiddleware)
app.use(cookieParser())
app.use(sessionMiddleware)
app.use(routes)
```

**Comment by maxpain** [comment]
> but it doesn't seem right to me to change the request headers

The main goal of this package is to be as compatible with `express` as possible, isn't it?

**Comment by nigrosimone** [comment]
Yes! I made a PR in a few hours!

**Comment by dimdenGD** [comment]
Released 1.3.12 on NPM with fix

**Comment by maxpain** [comment]
thanks!
